### PR TITLE
🎨 Allow project node patch of service key 

### PR DESCRIPTION
**Pull request by matusdrobuliak66**
## What do these changes do?
- 🎨 Allow project node patch of service key 

## Related issue/s
- relates https://github.com/ITISFoundation/osparc-issues/issues/1404

## Dev-ops checklist

- [ ] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by odeimaiz — APPROVED** [review]
thanks

File: services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py [hunk 1]
```diff
@@ -168,6 +168,7 @@ async def create_node(request: web.Request) -> web.Response:
     return envelope_json_response(data, status_cls=web.HTTPCreated)
 
 
+# NOTE: Careful, this endpoint is actually "get_node_state," and it doesn't return a Node resource.
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py:171]: TIP: add this note below the decorator or inside get_node function. This way it does not get accidentally pushed to the top by the auto-formatting tools

File: services/web/server/src/simcore_service_webserver/projects/db.py [hunk 2]
```diff
@@ -876,6 +876,7 @@ async def remove_project_node(
         async with self.engine.acquire() as conn:
             await project_nodes_repo.delete(conn, node_id=node_id)
 
+    # NOTE: Not all Node data are here yet; they are in the workbench of a Project, waiting to be moved here.
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/db.py:879]: same here

File: services/web/server/src/simcore_service_webserver/projects/projects_api.py [hunk 3]
```diff
@@ -916,7 +917,27 @@ async def patch_project_node(
...
+            user_id=user_id, project_uuid=f"{project_id}"
+        )
+        _project_node_data = _project["workbench"][f"{node_id}"]
+
+        _service_key = _node_patch_exclude_unset.get("key", _project_node_data["key"])
+        _service_version = _node_patch_exclude_unset.get(
+            "version", _project_node_data["version"]
+        )
+        rabbitmq_rpc_client = get_rabbitmq_rpc_client(app)
+        await catalog_rpc.get_service(
```
Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/projects_api.py:932]: I can add a check function. `get_service`  is pretty expensive just to check that a service exists and the user has rights to it.

Comment (pcrespov) [inline services/web/server/src/simcore_service_webserver/projects/projects_api.py:932]: See in `packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/catalog/errors.py` the exceptions that this function can raise. How do you handle them?

File: services/web/server/tests/unit/with_dbs/02/test_projects_nodes_handlers__patch.py [hunk 4]
```diff
@@ -40,6 +40,15 @@ def mock_project_uses_available_services(mocker: MockerFixture):
     )
 
 
+@pytest.fixture
+def mock_catalog_rpc_get_service(mocker: MockerFixture):
+    mocker.patch(
+        "simcore_service_webserver.projects.projects_api.catalog_rpc.get_service",
+        spec=True,
```
Comment (pcrespov) [inline services/web/server/tests/unit/with_dbs/02/test_projects_nodes_handlers__patch.py:47]: You can also raise one of the exceptions in `packages/service-library/src/servicelib/rabbitmq/rpc_interfaces/catalog/errors.py` to check how your code responds to it

File: services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py [hunk 5]
```diff
@@ -172,6 +172,7 @@ async def create_node(request: web.Request) -> web.Response:
 @login_required
 @permission_required("project.node.read")
 @_handle_project_nodes_exceptions
+# NOTE: Careful, this endpoint is actually "get_node_state," and it doesn't return a Node resource.
```
Comment (sanderegg) [inline services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py:175]: not sure I fully understand the note here.

**Review by sanderegg — APPROVED** [review]
thanks!

File: services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py (hunk 5 above)
Comment (matusdrobuliak66) [reply services/web/server/src/simcore_service_webserver/projects/_nodes_handlers.py:175]: `@routes.get(f"/{VTAG}/projects/{{project_id}}/nodes/{{node_id}}", name="get_node")` on this endpoint I would expect to receive the node resource with all node fields, but this is more like /projects/<project-id>/nodes/<node-id>/state endpoint, where you get just a state of a node.

**Review by GitHK — APPROVED** [review]

**Review by pcrespov — APPROVED** [review]
thx

**Pull request merged**